testsuite/tests/kuadrant/authorino/response/test_deny_with.py

```diff
@@ -0,0 +1,63 @@
+"""Test for custom deny responses."""
+import pytest
+
+from testsuite.objects import Value, ValueFrom, Rule
+
+HEADERS = {
+    "x-string-header": Value("abc"),
+    "x-int-header": Value(16),
+    "x-list-header": Value([1, 2, 3]),
+    "x-dict-header": Value({"anything": "something"}),
+    "x-dynamic-header": ValueFrom("context.request.http.path"),
+}
+
+TESTING_PATH = "/deny"
+
+
+@pytest.fixture(scope="module")
+def authorization(authorization):
+    """Set custom deny responses and auth rule with only allowed path '/allow'"""
+    authorization.responses.set_deny_with(
+        "unauthenticated",
+        code=333,
+        headers=HEADERS,
+        message=Value("Unauthenticated message"),
+        body=Value("You are unauthenticated."),
+    )
+    authorization.responses.set_deny_with(
+        "unauthorized",
+        code=444,
+        headers=HEADERS,
+        message=ValueFrom("My path is: " + "{context.request.http.path}"),
+        body=ValueFrom("You are unauthorized from path: " + "{context.request.http.path}"),
+    )
+    # Authorize only when url path is "/allow"
+    authorization.authorization.add_auth_rules("Whitelist", [Rule("context.request.http.path", "eq", "/allow")])
+    return authorization
+
+
+def check_headers(response):
+    """Check deny headers with normalization between HTTP strings and Python strings."""
+    assert response.headers["x-string-header"] == HEADERS["x-string-header"].value
+    assert response.headers["x-int-header"] == str(HEADERS["x-int-header"].value)
+    assert response.headers["x-list-header"] == str(HEADERS["x-list-header"].value).replace(" ", "")
+    assert response.headers["x-dict-header"] == str(HEADERS["x-dict-header"].value).replace(" ", "").replace("'", '"')
```

Comment:
Arent this just a json representations? I think trying to parse them from string to JSON objects and then comparing them would be better

Reply:
Using `json.loads` for comparison now